### PR TITLE
Remove file-based function resolution from call lowering

### PR DESCRIPTION
**Pull request by mcereal**
This PR removes the file-based function resolution mechanism that was previously used during LLVM call lowering. The system relied on serializing context function metadata to individual files in `build/sailfin/` and reading them back during call resolution, which added unnecessary I/O overhead and complexity.

## Key Changes

- **Removed file-based function lookup**: Deleted `_parse_int_res()`, `_find_function_from_files_res()`, and the entire file-based resolution block from `resolve_call_signature()` in `core_call_resolution.sfn`
- **Simplified function resolution**: Updated `_find_function_by_name_or_import()` to remove file I/O dependency and rely solely on in-memory function lookups
- **Removed serialization infrastructure**: Deleted `serialize_context_functions()` from `lowering_phase_types.sfn` that was writing function metadata to disk
- **Cleaned up imports**: Removed unused imports (`NativeParameter`, `char_at`, `char_code`, `map_parameter_type`) that were only needed for file-based resolution
- **Updated call sites**: Removed the call to `serialize_context_functions()` from `lowering_core.sfn`
- **Cleanup utilities**: Updated `_clean_lowering_state()` in `cli_commands.sfn` to remove references to `.ctx_func_*` scratch files

## Implementation Details

The file-based resolution was attempting to handle cases where function metadata might be corrupted during ABI processing. By removing this mechanism, the compiler now relies entirely on in-memory function lookups, which is simpler and more efficient. The `_find_function_by_name_or_import()` function still supports suffix-based matching for method resolution but does so without file I/O.

https://claude.ai/code/session_016pDR1mxB5DPs8cNiMDzJeb

### CODE REVIEW

File: compiler/src/llvm/lowering/lowering_core.sfn [hunk 1]
```diff
@@ -507,8 +507,6 @@ fn lower_to_llvm_lines_with_parsed_context(native_module: NativeModule, parse_in
     context_functions = extend_native_functions_inplace(context_functions, imported_struct_methods);
     if debug_lowering { print.err("emit-llvm: phase=context_functions"); }
```
[the remark this answers is not in the document]
Comment (mcereal) [reply compiler/src/llvm/lowering/lowering_core.sfn:508]: Skipping a new dedicated test on this PR. Rationale:

- Call signature resolution is exercised exhaustively by the existing test suite — every test that compiles a function (unit/integration/e2e) flows through `resolve_call_signature`. `make rebuild` itself is the strongest regression test, since the self-hosted compiler resolves ~thousands of cross-module call sites (imported functions, async, struct methods) on every build. If the in-memory `functions` parameter were missing anything the channel had, `make rebuild` would fail immediately.
- Per the procedure in `docs/build-performance.md` §259 step 7: "Rebuild (`make rebuild`) to confirm self-hosting still succeeds. This is the critical gate" — that's the contract for IPC channel removals on this Phase 2 track. Adding a parallel synthetic test would duplicate coverage already provided by self-hosting.
- Project guidance is to keep IPC-removal PRs surgical (one channel, one PR). A new test capsule would expand scope.

If `make test` (CI's Build+Test job) regresses, that's the signal. I'll watch the CI status and report.

---
_Generated by [Claude Code](https://claude.ai/code)_
